rust_src/src/eval.rs

```diff
@@ -1338,4 +1338,143 @@ pub fn condition_case(args: LispCons) -> LispObject {
     unsafe { internal_lisp_condition_case(var, bodyform, handlers) }
 }
 
+type SpecbindingRef = ExternalPtr<specbinding>;
+
```

Comment:
We have made similar functions into member functions of the type. Does that makes sense to do for `SpecBindingRef` as well? Look at `LispBufferRef` as an example.

Reply:
> We have made similar functions into member functions of the type.

Can you explain about what functions you talking? If about `specpdl_*`, I made them as functions because they used from C code, so I don't think that make them into methods make sense.

Reply:
Use by C code *today*. As we port the code to Rust will they continue to be used as functions or could we be operating on the specpdl type itself? It is a common pattern in the port to have `extern "C"` functions which act as wrappers around the Rust types and methods.

Reply:
Hm, you right. I'm gonna do that a little bit later.

Reply:
What we have been aiming for is the smallest possible `extern "C"` functions which simply translate the C needs into Rust and the Rust result back into C. This way the Rust code can use the desired Rust idioms and style. This ends up looking like:

```
pub extern "C" fn foo(buffer: LispObject) -> LispObject {
    let result = foo_rust(buffer.into());
    result.into()
}

pub fn foo_rust(buffer: LispBufferRef) -> int32 {
    // all of the original C logic here
}
```

Reply:
Okay, I made `specpdl_*` into methods of `SpecbindingRef` type and moved `SpecbindingRef` (and its impl) to `lisp.rs`, because `specbinding` originally was defined in `lisp.h`. I also, as you suggested, split `default_toplevel_binding` into `_rust` and `extern "C"` functions.

Reply:
@shaleh Hi. Can you review changes again?